src/actions/updateThoughts.ts

```diff
@@ -186,8 +188,34 @@ const updateThoughts = (
   const lexemeIndexOld = { ...state.thoughts.lexemeIndex }
   const lexemeIndexUpdatesOld = keyValueBy(lexemeIndexUpdates, key => ({ [key]: lexemeIndexOld[key] }))
 
-  // TODO: Can we use { overwritePending: !local } and get rid of the overwritePending option to updateThoughts? i.e. Are there any false positives when local is false?
-  const thoughtIndex = mergeUpdates(thoughtIndexOld, thoughtIndexUpdates, { overwritePending })
```

Comment:
Notice how the previous code had a very neat two calls to `mergeUpdates`, one after another. We've lost that nice symmetry in the new code. Let's factor the added logic into a helper function to avoid making `updateThoughts` messier.

Reply:
I've brought back the symmetry although not as nice as it was because we now need to pass `clearPending` here.

Reply:
I'm not sure that injecting `clearPending` is a good tradeoff here. Some structural symmetry would help make it more readable, but I don't want to complicate the API.

Also see: https://github.com/cybersemics/em/pull/3133#discussion_r2261764185